File: src/main/java/com/wyun/thrift/server/business/BusinessServiceMap.java
package com.wyun.thrift.server.business;

import com.alibaba.fastjson.JSONObject;
import com.wyun.thrift.server.Response;

import java.util.HashMap;
import java.util.Map;

/**
 * User:lpf
 * Date:2017/6/10
 * Time:16:52
 */
public class BusinessServiceMap {
    public static Map<String, IBusinessService> serviceMap = new HashMap();

    public Response handle(String serviceName, String operation, JSONObject request) {
        IBusinessService businessService=serviceMap.get(serviceName);
        Response result=businessService.handle(operation,request);
        return result;
    }

    public void registerService(String serviceName,IBusinessService businessService){
        serviceMap.put(serviceName,businessService);
    }
}

File: business/business_service_map.go
package business

import "code.aliyun.com/wyunshare/thrift-server/gen-go/server"

type BusinessServiceMap struct {
	ServiceMap map[string]IBusinessService
}
func (b *BusinessServiceMap) Handle(serviceName string, operation string, paramJSON []byte) (*server.Response, error) {
	businessService := b.ServiceMap[serviceName]
	return businessService.Handle(operation, paramJSON)
}

func (b *BusinessServiceMap) RegisterService(serviceName string, businessService IBusinessService) {
	b.ServiceMap[serviceName] = businessService
}